### PR TITLE
Add example on artifact detection by Riemannian potato

### PR DESCRIPTION
**Pull request by qbarthelemy**
This PR:

- adds the function `predict_proba` to `Potato` class (related to #45), useful for a future `PotatoField` class,

- adds an animated example in the gallery, on online artifact detection by Riemannian potato, with a comparison with Euclidean potato.

### CODE REVIEW

**Review by agramfort — COMMENTED** [review]
let's merge this one https://github.com/pyRiemann/pyRiemann/pull/106 first so we can see how the example renders before merging.

thx @qbarthelemy

**Comment by agramfort** [comment]
I would use a context manager
https://numpy.org/devdocs/reference/generated/numpy.testing.suppress_warnings.html
to avoid a global change of numpy behavior. If people copy paste this code
it can have side effets

>

File: examples/artifacts/plot_detect_riemannian_potato_EEG.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
+"""
+===============================================================================
+Artifact Detection with Riemannian Potato
+===============================================================================
+
+Example of Riemannian Potato [1] applied on EEG time-series to detect artifacts
```
Comment (agramfort) [inline examples/artifacts/plot_detect_riemannian_potato_EEG.py:6]: the link to reference does not render well.

File: examples/artifacts/plot_detect_riemannian_potato_EEG.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+import os
+import numpy as np
+from mne.datasets import sample                    # tested with mne 0.21
+from mne.io import read_raw_fif
+from mne import Epochs, make_fixed_length_events
+from pyriemann.estimation import Covariances
+from pyriemann.clustering import Potato
+from matplotlib import pyplot as plt
+from matplotlib.colors import to_rgb
+from matplotlib.animation import FuncAnimation
```
Comment (agramfort) [inline examples/artifacts/plot_detect_riemannian_potato_EEG.py:25]:
```suggestion
import os

import numpy as np
from matplotlib import pyplot as plt
from matplotlib.colors import to_rgb
from matplotlib.animation import FuncAnimation

from mne.datasets import sample
from mne.io import read_raw_fif
from mne import Epochs, make_fixed_length_events

from pyriemann.estimation import Covariances
from pyriemann.clustering import Potato
```

File: examples/artifacts/plot_detect_riemannian_potato_EEG.py [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+# Apply common average reference on EEG channels
+raw.pick_types(meg=False, eeg=True).apply_proj()
+
+# Select two EEG channels for the example, preferably without artifact at the
+# beginning to have a reliable calibration
+ch_names = ['EEG 010', 'EEG 015']
+
+# Apply band-pass filter between 1 and 35 Hz
+raw.filter(1., 35., method='iir', picks=ch_names)
```
Comment (agramfort) [inline examples/artifacts/plot_detect_riemannian_potato_EEG.py:90]: any reason to use a FIR and not an IIR that mne uses by default?

File: examples/artifacts/plot_detect_riemannian_potato_EEG.py [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+
+# Epoch time-series with a sliding window
+duration = 2.5    # duration of epochs
+interval = 0.2    # interval between epochs
+epochs = Epochs(raw, make_fixed_length_events(raw, id=1, duration=interval),
+                tmin=0., tmax=duration, baseline=None, verbose=False)
+epochs_data = 5e5 * epochs.get_data(picks=ch_names)
```
Comment (agramfort) [inline examples/artifacts/plot_detect_riemannian_potato_EEG.py:97]: you can now use mne.make_fixed_length_epochs

https://mne.tools/dev/generated/mne.make_fixed_length_epochs.html

File: examples/artifacts/plot_detect_riemannian_potato_EEG.py [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+z_th = 2.5       # z-score threshold
+t = 40           # nb of matrices to train the potato
+
+# Calibrate potato by unsupervised training on first matrices: compute a
+# reference matrix, mean and standard deviation of distances to this reference.
+train_set = range(t)
+rp = Potato(metric='riemann', threshold=z_th).fit(covs[train_set])
+rp_center = rp._mdm.covmeans_[0]
+ep = Potato(metric='euclid', threshold=z_th).fit(covs[train_set])
+ep_center = ep._mdm.covmeans_[0]
```
Comment (agramfort) [inline examples/artifacts/plot_detect_riemannian_potato_EEG.py:116]: can you use more explicit names that rp and ep. thx

File: examples/artifacts/plot_detect_riemannian_potato_EEG.py [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+plot_potato_2D(axs[1], None, X, Y, ep_zscores, ep_center, covs[train_set],
+               ep_colors, 'Z-score of Euclidean distance to reference')
+plt.show()
+
+
+###############################################################################
+# Online Artifact Detection with Potato
+# -------------------------------------
+
+# Detect artifacts/outliers on test set, with an animation to imitate an online
```
Comment (agramfort) [inline examples/artifacts/plot_detect_riemannian_potato_EEG.py:155]:
```suggestion
# -------------------------------------
#
# Detect artifacts/outliers on test set, with an animation to imitate an online
```

File: pyriemann/clustering.py [hunk 7]
```diff
@@ -400,7 +401,32 @@ def predict(self, X):
...
+        computing the right-tailed probability from z-score.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_trials, n_channels, n_channels)
+            ndarray of SPD matrices.
+
+        Returns
+        -------
+        proba : ndarray, shape (n_trials)
```
Comment (agramfort) [inline pyriemann/clustering.py:416]:
```suggestion
        proba : ndarray, shape (n_trials,)
```

File: pyriemann/clustering.py [hunk 8]
```diff
@@ -400,7 +401,32 @@ def predict(self, X):
         out[pred] = self.pos_label
         return out
 
+    def predict_proba(self, X):
+        """probability of being clean / belonging to the potato.
+        It is the probability to reject the null hypothesis "clean data",
+        computing the right-tailed probability from z-score.
```
Comment (agramfort) [inline pyriemann/clustering.py:407]:
```suggestion
        """Probability of being clean / belonging to the potato.

        It is the probability to reject the null hypothesis "clean data",
        computing the right-tailed probability from z-score.
```

File: examples/artifacts/plot_detect_riemannian_potato_EEG.py (hunk 4 above)
Comment (qbarthelemy) [reply examples/artifacts/plot_detect_riemannian_potato_EEG.py:97]:
```
""" Test make_fixed_length_events vs make_fixed_length_epochs """

import os
from matplotlib import pyplot as plt
from mne.datasets import sample
from mne.io import read_raw_fif
from mne import Epochs, make_fixed_length_events, make_fixed_length_epochs

###############################################################################
# Load EEG data
# -------------

raw_fname = os.path.join(sample.data_path(), 'MEG', 'sample',
                         'sample_audvis_filt-0-40_raw.fif')
raw = read_raw_fif(raw_fname, preload=True, verbose=False)
sfreq = int(raw.info['sfreq']) # 150 Hz

# Keep 2 EEG channels
raw.pick_types(meg=False, eeg=True).pick_channels(['EEG 001', 'EEG 010'])
raw.filter(1., 35.)
print(raw.info)
raw_shape = raw.get_data().shape 
# shape = 2 channels and 41700 samples (or 278 seconds)

# Epoch EEG time-series with a sliding window
length = 2.5      # length of window = length of epochs
interval = 0.2    # interval / overlap between consecutive epochs

# So, we must approximatively (without taking account side effects)
# obtain: 278/0.2 = 1390 epochs

###############################################################################
# Solution 1: define an event for each epoch, then use Epoch
# ----------------------------------------------------------

# make_fixed_length_events: make a set of events separated by a fixed duration.
# In our case, overlaped epochs are separated by a duration equal to interval.
events = make_fixed_length_events(raw, id=1, duration=interval)
print("shape of events =", events.shape)
# shape = (1388, 3) => OK

# Plot events on EEG raw signal
fig_r = raw.plot(events=events, duration=8, start=0)
plt.savefig("fig_r")

# Epoch EEG on events, from 0 to 2.5 seconds
epochs1 = Epochs(raw, events, tmin=0., tmax=length, baseline=None)
print("shape of epochs1 =", epochs1.get_data().shape)
# shape = (1377, 2, 376) => OK

# Plot epochs1
fig_e1 = epochs1.plot(n_epochs=5)
plt.savefig("fig_e1")

###############################################################################
# Solution 2: use make_fixed_length_epochs
# ----------------------------------------

# make_fixed_length_epochs: epoch data into equal-sized consecutive epochs.
# duration: duration of each epoch in seconds = length
# overlap: the overlap between epochs, in seconds = interval
epochs2 = make_fixed_length_epochs(raw, duration=length, overlap=interval,
                                   reject_by_annotation=False)
print("shape of epochs2 =", epochs2.get_data().shape)
# shape = (120, 2, 375) => not the good number of epochs => KO

# Plot epochs2
fig_e2 = epochs2.plot(n_epochs=5)
plt.savefig("fig_e2")
# Epochs are not overlapped => KO
```

**Review by agramfort — APPROVED** [review]

**Pull request merged**